pylint/testutils/output_line.py

```diff
@@ -62,39 +76,42 @@ def __init__(self, row, exception):
 class OutputLine(NamedTuple):
     symbol: str
     lineno: int
-    column: str
-    object: Any
+    column: int
+    object: str
     msg: str
-    confidence: interfaces.Confidence
+    confidence: str
 
     @classmethod
-    def from_msg(cls, msg):
+    def from_msg(cls, msg: Message) -> "OutputLine":
         column = cls.get_column(msg.column)
         return cls(
             msg.symbol,
             msg.line,
             column,
             msg.obj or "",
             msg.msg.replace("\r\n", "\n"),
-            msg.confidence.name
-            if msg.confidence != interfaces.UNDEFINED
-            else interfaces.HIGH.name,
+            msg.confidence.name if msg.confidence != UNDEFINED else HIGH.name,
         )
 
     @classmethod
-    def get_column(cls, column):
-        if not PY38_PLUS:
-            return ""  # pragma: no cover
-        return str(column)
+    def get_column(cls, column: str) -> int:
+        if (
+            not PY38_PLUS
+        ):  # We check the column only for the new better ast parser introduced in python 3.8
+            return 0  # pragma: no cover
+        return int(column)
 
     @classmethod
-    def from_csv(cls, row):
+    def from_csv(cls, row: Union[Tuple[str, ...], List[str], str]) -> "OutputLine":
         try:
-            confidence = row[5] if len(row) == 6 else interfaces.HIGH.name
             column = cls.get_column(row[2])
-            return cls(row[0], int(row[1]), column, row[3], row[4], confidence)
+            if isinstance(row, Iterable) and len(row) == 6:
```

Comment:
`isinstance(row, Iterable)` doesn't test if `row` supports `__len__`. You need at least a `Collection` or `Sequence` for that. Since you also use `__getitem__` -> `Sequence`.